### ISSUE TITLE
Document best practices for complex forms

### ISSUE THREAD

**Issue by mrjones-plip** [medic/cht-docs#308]
**Is your feature request related to a problem? Please describe.**
While working on the [CMoHM app](https://github.com/medic/config-moh-mali), an assessment [form](https://github.com/medic/config-moh-mali/blob/master/covid-surveillance/forms/app/assessment.xml) was built. The desire was not have the CHW to fill out multiple forms for the household, but to collapse all of this into one from.  As a result, the form can become [increasingly slower](https://github.com/medic/config-moh-mali/issues/56) for larger and larger households. 

**Describe the solution you'd like**
We should attempt to quantify the upper bounds of what the cht-core can handle in regards to this complexity in forms. There may be other limits that the core has that app builds don't know, so we should be sure ask what other issues builders have had.

**Describe alternatives you've considered**
[See the issue](https://github.com/medic/config-moh-mali/issues/56), but some work arounds in the form have been done to let the CHW know the app will be slow.

**Additional context**
This may end up in the cht-docs repo instead, but @abbyad had suggested I start here and move on.

FYI @kennsippell , @PhilipNgari , @derickl & @binokaryg

**Comment by mrjones-plip** [comment]
@abbyad  - does this look fleshed out enough based on our Tuesday chat?

**Comment by ecsalomon** [comment]
@mrjones-plip Would there be a use case for synthetic data here? We are initiating a project on synthetic data, and I am collecting use cases.

**Comment by mrjones-plip** [comment]
@ecsalomon  - oh - fascinating!  The details are not yet fleshed out for this ticket, but I suspect we'll want a static data set that app developers can repeatedly run through their forms to see how performant it is or isn't as the tweak things. Does this sound like a match for your project?

**Comment by ecsalomon** [comment]
Yeah, totally! I may ask you for some follow up details as we move this project forward. :)

**Comment by mrjones-plip** [comment]
Awesome! Happy to help as best as I newbie can, but will also defer a lot to @kennsippell and the rest of the App Services team.

**Comment by abbyad** [comment]
It will be difficult to "quantify the upper bounds of what the cht-core can handle" since it will often be limited by the device hardware, not by the CHT. The form mentioned in the subsequent paragraph to profile the performance would be helpful. It could be useful for that form to be dynamic in size, so that you can see how it performs on target devices as the elements are increased.

**Comment by mrjones-plip** [comment]
@abbyad  - cool cool, makes sense.  I've reworded the body of the ticket to reflect this.  Feedback welcome!
